homeassistant/components/template/weather.py

```diff
@@ -322,9 +332,23 @@ async def async_added_to_hass(self) -> None:
                 "_apparent_temperature",
                 self._apparent_temperature_template,
             )
+
         if self._forecast_template:
             self.add_template_attribute(
                 "_forecast",
                 self._forecast_template,
             )
+            self.async_on_remove(
+                async_track_state_change_event(
+                    self.hass, self.entity_id, self.async_state_changed_listener
+                )
+            )
```

Comment:
Instead of this, I think we can use the `on_update` parameter to `add_template_attribute`.